datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -238,13 +238,12 @@ pub struct ConstEvaluator {
     /// descendants) so this Expr can be evaluated
     can_evaluate: Vec<bool>,
 
-    ctx_state: ExecutionContextState,
```

Comment:
Here is reason 1 for this change: We no longer have to create a whole new `ctx_state` and `planner` for each potential constant evaluation 🎉